### PR TITLE
protocol: adding error job to log error with custom protocols

### LINKED ISSUE #1189: Error feedback with custom protocol handler
Is there any way we can send back errors with custom protocols?
E.g. when i access a file that doesn exists with the ```file://``` protocoll, chrome does know the file does not exist (in the devConsole this is logged as ```net::ERR_FILE_NOT_FOUND```)
With my custom ```foobar://``` protocoll, i want to look up files, but trigger errors when specific (e.g. forbidden) files are accessed.

The only current way i can think of is redirecting the call to a file that does not exist for sure:
```
return new protocol.RequestFileJob('#');
```
but a specifically triggered error (event with "Forbiden" or "Access Denied" messages) would be much nicer.

I guess it shouldn't be hard to implement, as error feedback already exists [here](https://github.com/atom/atom-shell/blob/master/atom/browser/api/atom_api_protocol.cc#L109).

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #1189 

screen shot 2015-05-07 at 1 15 17 pm

### CODE REVIEW

File: atom/browser/api/atom_api_protocol.cc [hunk 1]
```diff
@@ -105,6 +105,14 @@ class CustomProtocolRequestJob : public AdapterRequestJob {
             base::Bind(&AdapterRequestJob::CreateFileJobAndStart,
                        GetWeakPtr(), path));
         return;
+      } else if (name == "RequestErrorJob") {
+        int error;
```
Comment (zcbenz) [inline atom/browser/api/atom_api_protocol.cc]: We should have a default error here in case user didn't specify one.

Comment (deepak1556) [inline atom/browser/api/atom_api_protocol.cc]: Ah yeah, but what would be the default ?

Comment (zcbenz) [inline atom/browser/api/atom_api_protocol.cc]: Maybe `net::ERR_NOT_IMPLEMENTED`.

Comment (deepak1556) [inline atom/browser/api/atom_api_protocol.cc]: or `net::ERR_TIMED_OUT` ? will add the default value to the docs..

Comment (zcbenz) [inline atom/browser/api/atom_api_protocol.cc]: `net::ERR_TIMED_OUT` makes it look like a connection error.

Comment (deepak1556) [inline atom/browser/api/atom_api_protocol.cc]: cool, added `net::ERR_NOT_IMPLEMENTED` as default.

**Comment by zcbenz** [comment]
Thanks!

**Comment by Wortex17** [comment]
Thanks a lot! :+1:

**Pull request merged**